From 347ed661de839b2b4849d0ee41d2a10970b1d76d Mon Sep 17 00:00:00 2001
From: Christian Ehrhardt <christian.ehrhardt@codasip.com>
Date: Tue, 4 Mar 2025 09:37:34 +0100
Subject: qemu: Increase timer threshold

On qemu with CHERI most of the sleep tests fail because
wakeup takes too long on average. However, this is caused
by qemu's slowness. So increase the margine generously.

Upstream-Status: Inappropriate [ CHERI Specific ]

diff --git a/lib/tst_timer_test.c b/lib/tst_timer_test.c
index 512edc54b..3e941923a 100644
--- a/lib/tst_timer_test.c
+++ b/lib/tst_timer_test.c
@@ -188,8 +188,8 @@ static long long compute_threshold(long long requested_us,
 
 	slack_per_scall = MAX(slack_per_scall, timerslack);
 
-	return (400 + 2 * monotonic_resolution + slack_per_scall) * nsamples
-		+ 3000/nsamples;
+	return 10 * ((400 + 2 * monotonic_resolution + slack_per_scall) * nsamples
+		+ 3000/nsamples);
 }
 
 /*
-- 
2.43.5

